lib/ansible/plugins/inventory/tower.py

```diff
@@ -0,0 +1,201 @@
+                        raise AnsibleParserError("Tower server licenses must match: source: {source} local: {local}".
+                                                 format(source=source_type, local=license_type))
+                else:
+                    raise AnsibleParserError("Failed to validate the license of the remote Tower: {response}".
+                                             format(response=config_response))
+
+            response = requests.get(inventory_url,
+                                    auth=HTTPBasicAuth(tower_user, tower_pass),
+                                    verify=verify_ssl)
+            if not response.ok:
+                # If the GET /api/v2/inventories/N/script is not HTTP 200, print the error code
+                raise AnsibleParserError("Connection to remote host failed: {response}".format(response=response))
+            try:
+                # Attempt to parse JSON
+                return response.json()
+            except (ValueError, TypeError) as e:
+                # If the JSON parse fails, print the ValueError
+                raise AnsibleParserError("Failed to parse json from host: {err}".format(err=e))
+        except requests.ConnectionError as e:
+            raise AnsibleParserError("Connection to remote host failed: {err}".format(err=e))
+
+    def verify_file(self, path):
+        if path.endswith('tower_inventory'):
+            self.read_all_configs_from_env_var = True
+            return True
+        elif super(InventoryModule, self).verify_file(path):
+            return path.endswith('tower_inventory.yml') or path.endswith('tower_inventory.yaml')
+        else:
+            return False
+
+    def get_option(self, option, hostvars=None):
+        '''
+            The default get_option cannot cast int to string, even if the type is set.
+            We override that behavior and force all numbers into string.
+        '''
+        raw_value = super(InventoryModule, self).get_option(option)
+        if type(raw_value) == int:
+            return to_text(raw_value, errors='surrogate_or_strict')
+        else:
+            return raw_value
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+        if not self.read_all_configs_from_env_var and os.path.isfile(path):
+            self._read_config_data(path)
+        # Read inventory from tower server.
+        # Note the environment variables will be handled automatically by InventoryManager.
+        inventory = self.read_tower_inventory(self.get_option('tower_host'),
+                                              self.get_option('tower_username'),
+                                              self.get_option('tower_password'),
+                                              self.get_option('tower_inventory'),
+                                              os.environ.get('TOWER_LICENSE_TYPE', 'enterprise'),
```

Comment:
why is this ONLY an env var?

Reply:
I would really like to pull @matburt into the decision-making on this matter. It's not just the sourcing from the env var here, but also the use of the `license_type` parameter once it's received. What does this mean in practice? I think users would expect to be able to use the plugin as a part of the Ansible core CLI with any given type of target AWX or Tower instance.